WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java

```diff
@@ -219,14 +219,42 @@ private void startActivityWithSyntheticBackstack(@NonNull Intent intent) {
     }
 
     private void bumpAnalytics(ShareAction shareAction, int selectedSiteLocalId) {
+        SiteModel selectedSite = mSiteStore.getSiteByLocalId(selectedSiteLocalId);
+        int numberOfMediaShared = countMedia();
+
         Map<String, Object> analyticsProperties = new HashMap<>();
-        analyticsProperties.put("number_of_media_shared", countMedia());
+        analyticsProperties.put("number_of_media_shared", numberOfMediaShared);
         analyticsProperties.put("share_to", shareAction.analyticsName);
 
         AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.SHARE_TO_WP_SUCCEEDED,
-                mSiteStore.getSiteByLocalId(selectedSiteLocalId),
+                selectedSite,
                 analyticsProperties);
 
+        if (doesContainMediaAndWasSharedToMediaLibrary(shareAction, numberOfMediaShared)) {
+            trackMediaAddedToMediaLibrary(selectedSite);
+        }
+    }
+
+    private void trackMediaAddedToMediaLibrary(SiteModel selectedSite) {
+        ArrayList<Uri> mediaUrls = new ArrayList<>();
+        if (countMedia() == 1) {
+            Uri singleMedia = getIntent().getParcelableExtra(Intent.EXTRA_STREAM);
+            mediaUrls.add(singleMedia);
+        } else {
+            ArrayList<Uri> imageUris = getIntent().getParcelableArrayListExtra(Intent.EXTRA_STREAM);
+            mediaUrls.addAll(imageUris);
+        }
+
+        for (Uri uri : mediaUrls) {
+            String mimeType = getContentResolver().getType(uri);
+            boolean isVideo = mimeType != null && mimeType.startsWith("video");
+            Map<String, Object> properties = AnalyticsUtils.getMediaProperties(this, isVideo, uri, null);
+            AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.MEDIA_LIBRARY_ADDED_PHOTO, selectedSite, properties);
```

Comment:
I think we should check if it's a video or not here and bump stats accordingly:
```
if (isVideo) {
 AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.MEDIA_LIBRARY_ADDED_VIDEO, selectedSite, properties);
        } else {
 AnalyticsUtils.trackWithSiteDetails(AnalyticsTracker.Stat.MEDIA_LIBRARY_ADDED_PHOTO, selectedSite, properties);
        }
```

Reply:
> I think we should check if it's a video or not here and bump stats accordingly:

Great call @daniloercoli, addressed in f63d706